### ISSUE TITLE
[SUGGESTION] Option to override URL used for in-game links

### ISSUE THREAD

**Issue by BlazeGuy2** [StrangeLoopGames/EcoSuggestions#94]
_From @shadow7412 on April 27, 2018 1:39_

I'm hosting an eco server, and have DNS set up for it. However when people click on in-game links, instead of being taken to my neat https://eco.example.com they are taken instead to http://1.2.3.4:19337

I would like, if possible, a server setting that overrides the base url for all such interactions.
Ideally, this setting should reside inside a file so that it can be easily edited (as I'm running the server on a headless ubuntu, I don't have access to the UI).

Lastly, if left blank, it should probably default to its current behaviour.

_Copied from original issue: StrangeLoopGames/EcoIssues#7450_

**Comment by BlazeGuy2** [comment]
_From @Elliander on April 27, 2018 19:38_

Yeah, I noticed this as well. When I host, I see a local IP address and would prefer it if I could designate an alternative location - such as a domain I own.

**Comment by BlazeGuy2** [comment]
_From @Chi-Yu on May 11, 2018 23:17_

Seeing that browser developers are working on warning users more and more about insecure connections and using non-default ports also sometimes triggers security warnings, I'd really like to see this feature implemented.

Also, there are servers behind NAT which reportedly build those links with incorrect IP addresses from the local address space such as `192.168.*.*` which of course won't work outside the local network.

Implementing this feature request should also help those people with broken NAT configurations.

**Comment by BlazeGuy2** [comment]
_From @aawilson on May 18, 2018 22:12_

There was a mention in the 7.4.5 update around this, but no mention of any config, has anyone experimented with it and found out what mechanism they used to, quote, "[fix] webserver link from using the wrong address"?

**Comment by Chi-Yu** [comment]
As long as the server reports an appropriate IP address and port which work in the player's browser (firewall issues aside), why would it be a bug? It's merely a missing feature that a specific group of people would like to see.

**Comment by aawilson** [comment]
For one thing, it doesn't necessarily do that. There are plenty of setups where the IP address accessible to the player is not calculable server-side, thanks to the nature of web. The server I play on has this problem, even with the "fix" in 7.4.5: It uses the IP address that the A record points to, and thus does use the IP address that the users would connect to, but the server in question is a vhost setup that uses the Host header to distinguish between the vhosts and proxy appropriately. When it receives the bare IP, it serves a default vhost, and not Eco. This is typically resolved by either accepting a Host header equivalent, or by allowing a specific hostname to be specified in server config.

**Comment by BurningEnlightenment** [comment]
I would also like to see this, because I have an nginx acting as TLS reverse proxy in front of my dockerized eco server.

@SeanAtr0n @metachronism do you think this would be a suitable first contribution task? Would you accept a pr regarding this?

**Comment by shadow7412** [comment]
I'm thinking about starting up another eco server. Has there been any thoughts about this, or has it fallen to the bottom of the backlog?

**Comment by mirasrael** [comment]
It is already implemented as WebServerUrl in server network settings

**Comment by shadow7412** [comment]
Yep, I just confirmed this. There's no mention of it currently in Network.eco or Network.eco.template. I'm not sure if it's on the UI because I'm running the server in a headless linux server. Thanks :)
